central/sensor/service/connection/connection_test.go

```diff
@@ -442,6 +442,68 @@ func (s *testSuite) TestIssueLocalScannerCerts() {
 	}
 }
 
+func (s *testSuite) TestIssueSecuredClusterCerts() {
+	namespace, clusterID, requestID := "namespace", "clusterID", "requestID"
+	testCases := map[string]struct {
+		requestID  string
+		namespace  string
+		clusterID  string
+		shouldFail bool
+	}{
+		"no parameter missing": {requestID: requestID, namespace: namespace, clusterID: clusterID, shouldFail: false},
+		"requestID missing":    {requestID: "", namespace: namespace, clusterID: clusterID, shouldFail: true},
+		"namespace missing":    {requestID: requestID, namespace: "", clusterID: clusterID, shouldFail: true},
+		"clusterID missing":    {requestID: requestID, namespace: namespace, clusterID: "", shouldFail: true},
+	}
+	for tcName, tc := range testCases {
+		s.Run(tcName, func() {
+			sendC := make(chan *central.MsgToSensor)
+			sensorMockConn := &sensorConnection{
+				clusterID: tc.clusterID,
+				sendC:     sendC,
+				stopSig:   concurrency.NewErrorSignal(),
+				sensorHello: &central.SensorHello{
+					DeploymentIdentification: &storage.SensorDeploymentIdentification{
+						AppNamespace: tc.namespace,
+					},
+				},
+			}
+			ctx := context.Background()
+			ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
+			defer cancel()
+			request := &central.MsgFromSensor{
+				Msg: &central.MsgFromSensor_IssueSecuredClusterCertsRequest{
+					IssueSecuredClusterCertsRequest: &central.IssueSecuredClusterCertsRequest{
+						RequestId: tc.requestID,
+					},
+				},
+			}
+
+			handleDoneErrSig := concurrency.NewErrorSignal()
+			go func() {
+				handleDoneErrSig.SignalWithError(sensorMockConn.handleMessage(ctx, request))
+			}()
+
+			select {
+			case msgToSensor := <-sendC:
```

Comment:
nit: we are not checking if the expected certs are really returned, right?

Reply:
I now added some basic checks that data is returned.

Not sure about this:
```go
xpectedCertificates := 7
s.Equal(expectedCertificates, len(serviceCertificates))
```

Maybe it's too rigid, and we'll have to fix the test if the number of secured cluster services changes.

I wouldn't go much more in depth here, we test the certificate generation in `certificates_test.go`.